Source/JavaScriptCore/heap/WeakSet.h

```diff
@@ -115,9 +115,9 @@ inline bool WeakSet::isTriviallyDestructible() const
     return true;
 }
 
-inline void WeakSet::deallocate(WeakImpl* weakImpl)
+ALWAYS_INLINE void WeakSet::deallocate(WeakImpl* weakImpl)
 {
-    weakImpl->setState(WeakImpl::Deallocated);
+    weakImpl->clear();
```

Comment:
This is a behavior change: the old code preserves the handle owner pointer, the new code nullifies it.  Why is this ok?

Reply:
Discussed offline. This is intentional since WeakImpl gets deallocated. So owner is not touched anymore. This change avoids unnecessary load.